### ISSUE TITLE
Have lockfile in another directory than the flutter executable such as /tmp or ~/.cache

### ISSUE THREAD

**Issue by Alkeryn** [flutter/flutter#44526]
I want to package flutter for nix but i cannot do that due to it trying to write to its install directory, it would be nice for flutter to not attempt to write to the folder it is installed in as it make it very hard to package it correctly.

**Comment by christopherfujino** [comment]
@Alkeryn I agree! Unfortunately, this would be a non-trivial refactor. However, it definitely needs to be done, as it causes installation problems not only on Linux, but also Mac and Windows.
